File: python_code/oop/SOLID/bird/Parrot.py
from Bird import *
from BirdType import *
from interfaces.FlyableInterface import *
from interfaces.FlyingBehaviourInterface import *

class Parrot(Bird, FlyableInterface):
    __flyingBehaviour: FlyingBehaviourInterface = None
    def __init__(self, weight: int, colour: str, size: int, beakType: str, birdType: BirdType, flyingBehaviour: FlyingBehaviourInterface) -> None:
        super().__init__(weight, colour, size, beakType, birdType)
        self.__flyingBehaviour = flyingBehaviour
    
    def fly(self) -> None:
        self.__flyingBehaviour.makeFly()
        
    def makeSound(self) -> None:
        ...

File: oop/code/java/src/main/java/com/scaler/lld/bird/Parrot.java
package com.scaler.lld.bird;

public class Parrot extends Bird {

    public Parrot(Integer weight, String colour, String size, String beakType, BirdType type) {
        super(weight, colour, size, beakType, type);
    }

    @Override
    public void fly() {
        System.out.println("\nParrot is flying");
    }
    
}
